File: 0002_AddTwoNumbers.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode* out = new ListNode(NULL);
        ListNode* out_copy = out;
        int countup = 0;
        while ((l1 != NULL || l2 != NULL) || countup == 1){
            if(l1 != NULL && l2 != NULL) {
                out_copy->val = (l1->val + l2->val + countup) % 10;
                countup = (l1->val + l2->val + countup)/10 % 10;
            } else if (l1 == NULL && l2 != NULL) {
                out_copy->val = (l2->val + countup) % 10;
                countup = (l2->val + countup)/10 % 10;
            } else if (l1 != NULL && l2 == NULL) {
                out_copy->val = (l1->val + countup) % 10;
                countup = (l1->val + countup)/10 % 10;
            } else {
                out_copy->val = countup;
                countup = 0;
            }

            if (l1 != NULL) l1 = l1->next;
            if (l2 != NULL) l2 = l2->next;
            if((l1 != NULL || l2 != NULL) || countup == 1) {
                ListNode* next = new ListNode(NULL);
                out_copy->next = next;
                out_copy = out_copy->next;
            }   
        }
        return out;
    }
};

File: 0002_AddTwoNumbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
        updigit = 0
        ans_cur = ListNode(0, None)
        ans = ans_cur
        while l1 or l2:
            cur_sum = updigit
            if l1:
                cur_sum += l1.val
                l1 = l1.next
            if l2:
                cur_sum += l2.val
                l2 = l2.next
            updigit = cur_sum // 10
            ans_cur.val = cur_sum % 10
            if l1 or l2 or updigit > 0:
                ans_cur.next = ListNode(0, None)
                ans_cur = ans_cur.next
        if updigit > 0:
            ans_cur.val = updigit
        return ans
